### PR TITLE
fix(gatsby-transformer-remark): Check key before trying to get cache

### LINKED ISSUE #16703: gatsby-remark-images error if using markdownCaptions: true
## Description

`gatsby-remark-images` produces and error if using `markdownCaptions: true` in `gatsby-config.js`

### Steps to reproduce

1. Create a new Gatsby site using the gatsby-starter-blog (e.g gatsby new my-themed-blog https://github.com/gatsbyjs/gatsby-starter-blog-theme)
2. Change `gatsby-config.js` to add option `markdownCaptions: true` on `gatsby-remark-images` block
3. Run yarn develop

### Expected result

The page content should be generated and the development site should be available.

### Actual result

Errors on terminal

```
UNHANDLED REJECTION Cannot read property 'contentDigest' of undefined



  TypeError: Cannot read property 'contentDigest' of undefined

  - extend-node-type.js:57 htmlCacheKey
    [test]/[gatsby-transformer-remark]/extend-node-type.js:57:80

  - extend-node-type.js:337 Object.getHTML [as generateHTML]
    [test]/[gatsby-transformer-remark]/extend-node-type.js:337:42

  - index.js:150 getImageCaption
    [test]/[gatsby-remark-images]/index.js:150:21

  - index.js:306 _callee$
    [test]/[gatsby-remark-images]/index.js:306:54
```

### Environment

  System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Binaries:
    Yarn: 1.17.3
    npm: 6.10.3
  Languages:
    Python: 3.7.1
  Browsers:
    Edge: 44.18362.267.0
  npmPackages:
    gatsby: ^2.13.67 => 2.13.67 
    gatsby-image: ^2.2.8 => 2.2.8 
    gatsby-plugin-feed: ^2.3.6 => 2.3.6 
    gatsby-plugin-google-analytics: ^2.1.7 => 2.1.7 
    gatsby-plugin-manifest: ^2.2.5 => 2.2.5 
    gatsby-plugin-offline: ^2.2.6 => 2.2.6 
    gatsby-plugin-react-helmet: ^3.1.3 => 3.1.3 
    gatsby-plugin-sharp: ^2.2.12 => 2.2.12 
    gatsby-plugin-typography: ^2.3.2 => 2.3.2 
    gatsby-remark-copy-linked-files: ^2.1.6 => 2.1.6 
    gatsby-remark-images: ^3.1.12 => 3.1.12 
    gatsby-remark-prismjs: ^3.3.5 => 3.3.5 
    gatsby-remark-responsive-iframe: ^2.2.4 => 2.2.4 
    gatsby-remark-smartypants: ^2.1.2 => 2.1.2 
    gatsby-source-filesystem: ^2.1.9 => 2.1.9 
    gatsby-transformer-remark: ^2.6.14 => 2.6.14 
    gatsby-transformer-sharp: ^2.2.6 => 2.2.6 

### More info

It looks like the `gatsby-remark-images` generate caption from markdown `getHTML` function from `gatsby-transform-remark`. I guess The "markdownnode" of the image caption does not have `node.internal.contentDigest`

### PR DESCRIPTION
**Pull request by mathieudutour**
## Description

- [x] make sure it's something we want to cache before trying to get the cache of the markdown node

## Related Issues

Fixes #16703

### CODE REVIEW

**Comment by muescha** [comment]
I read there is `contentDigest` missing. Is this missing `contentDigest` a bug?

**Comment by mathieudutour** [comment]
It’s not really. The transformer passes the function to its plugins, which can call it to generate html. But when it’s called from plugins, it does not have the `internal` property.

**Comment by mathieudutour** [comment]
What's the deal with `Gatsby Build Service`? I can't seem to be able to see the logs (`[GraphQL error]: Message: Internal server error, Location: undefined, Path: undefined` in the console)

File: packages/gatsby-transformer-remark/src/extend-node-type.js [hunk 1]
```diff
@@ -330,7 +330,9 @@ module.exports = (
     }
 
     async function getHTML(markdownNode) {
-      const cachedHTML = await cache.get(htmlCacheKey(markdownNode))
+      const shouldCache = !!markdownNode && markdownNode.internal
```
Comment (sidharthachatterjee) [inline packages/gatsby-transformer-remark/src/extend-node-type.js]:
```suggestion
      const shouldCache = markdownNode && markdownNode.internal
```

Do we need the double _not_ operator here?

**Pull request merged**